### PR TITLE
[pwrmgr, top] Update documentation regarding flash / reset

### PR DESCRIPTION
**Pull request by tjaychen**
- clarify pwrmgr handling for both fall-through / abort
- Add references in top level documentation

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
does this look good enough to merge?

**Comment by tjaychen** [comment]
can someone help approve so this can be merged?

File: hw/ip/pwrmgr/doc/_index.md [hunk 1]
```diff
@@ -110,9 +110,22 @@ Specifically, this means if software issues only WFI, the power manager does not
 The notion of WFI is exported from the processor.
 For Ibex, this is currently in the form of `core_sleeping_o`.
 
-In response to the low power entry request, the fast FSM disables all second level clock gating and asserts appropriate resets if required.  The
-fast FSM then requests the slow FSM to take over.
-Once the request is acknowledged, the fast FSM transitions to low power and waits for the next power up request.
+In response to the low power entry request, the fast FSM disables all second level clock gating.
+Before proceeding, the fast FSM explicitly separates the handling between a normal low power entry and a [reset request](#reset-request-handlig).
```
[the remark this answers is not in the document]
Comment (matutem) [reply hw/ip/pwrmgr/doc/_index.md:114]: Well, this is not right, so it should be resolved.

**Review by matutem — APPROVED** [review]
Approving with some changes requested.

File: hw/ip/pwrmgr/doc/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tjaychen) [reply hw/ip/pwrmgr/doc/_index.md:114]: ah let me add some explanation if it's not there. The fall through handling was actually meant to catch the fall through interrupt case that occurs RIGHT when software issues WFI. I thought I had explained this, but maybe i didn't?

File: hw/top_earlgrey/doc/_index.md [hunk 2]
```diff
@@ -153,6 +153,39 @@ Resets throughout the design are asynchronous active low as per the Comportabili
...
+
+* Reset due to external supply lowering.
+* Reset due to internal peripheral request.
+* Reset due to lower power entry and exit.
+
+#### Reset due to External Supply
+
+When the device resets due the supply drop, this is commonly known as "brown-out".
+When this occurs, the flash memory must go through specialized sequencing to ensure the cells are not damaged.
+This process is handled exclusively between [ast]({{< relref "hw/top_earlgrey/ip/ast/doc" >}})) and the flash.
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply hw/top_earlgrey/doc/_index.md:171]: yes it should! sorry when the pwrmgr doc was first written, the ast link didn't exist yet, and i never remembered to update it.

File: hw/top_earlgrey/doc/_index.md [hunk 3]
```diff
@@ -153,6 +153,39 @@ Resets throughout the design are asynchronous active low as per the Comportabili
...
+#### Reset due to Internal Request
+
+When the device receives an internal request to reset (for example [aon_timer]({{< relref "hw/ip/aon_timer/doc/#aon-watchdog-timer" >}})), device power is kept on and the flash is directly reset.
+It is assumed that the flash device, when powered, will be able to correctly handle such a sequence and properly protect itself.
+
+#### Reset due to Low Power Entry
+
+When the device receives a low power entry request while flash activity is ongoing, the [pwrmgr]({{< relref "hw/ip/pwrmgr/doc/#abort-handling" >}})) is responsible for ensuring the entry request is aborted.
+
+
```
[the remark this answers is not in the document]
Comment (tjaychen) [reply hw/top_earlgrey/doc/_index.md:183]: will do.

**Pull request merged**